From 8920faeb9669ef802eb598da0c033d42cd450658 Mon Sep 17 00:00:00 2001
From: "Philip P. Moltmann" <moltmann@google.com>
Date: Wed, 28 Oct 2020 14:00:32 -0700
Subject: [PATCH 13/51] [Q_asb_2021-01] Hide overlays over uninstall confirm
 dialog

Test: atest CtsPackageUninstallTestCases
Fixes: 171221302
Merged-In: I38b6d85871064d76f2911e20acc74b4ab76406b3
Change-Id: I38b6d85871064d76f2911e20acc74b4ab76406b3
(cherry picked from commit d75105e74b6745f40caf872ea30fbc9d5ba8b81e)
---
 .../src/com/android/packageinstaller/UninstallerActivity.java  | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/packages/PackageInstaller/src/com/android/packageinstaller/UninstallerActivity.java b/packages/PackageInstaller/src/com/android/packageinstaller/UninstallerActivity.java
index be778e92787e..94829b506f95 100755
--- a/packages/PackageInstaller/src/com/android/packageinstaller/UninstallerActivity.java
+++ b/packages/PackageInstaller/src/com/android/packageinstaller/UninstallerActivity.java
@@ -17,6 +17,7 @@
 package com.android.packageinstaller;
 
 import static android.app.AppOpsManager.MODE_ALLOWED;
+import static android.view.WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS;
 
 import static com.android.packageinstaller.PackageUtil.getMaxTargetSdkVersionForUid;
 
@@ -87,6 +88,8 @@ public class UninstallerActivity extends Activity {
 
     @Override
     public void onCreate(Bundle icicle) {
+        getWindow().addSystemFlags(SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
+
         // Never restore any state, esp. never create any fragments. The data in the fragment might
         // be stale, if e.g. the app was uninstalled while the activity was destroyed.
         super.onCreate(null);
-- 
2.34.1

